examples/runners/wgpu/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+            }
+            Event::WindowEvent {
+                event: WindowEvent::Resized(size),
+                ..
+            } => {
+                // Recreate the swap chain with the new size
+                sc_desc.width = size.width;
+                sc_desc.height = size.height;
+                if let Some(surface) = &surface {
+                    swap_chain = Some(device.create_swap_chain(surface, &sc_desc));
+                }
+            }
+            Event::RedrawRequested(_) => {
+                if let Some(swap_chain) = &mut swap_chain {
+                    let frame = swap_chain
+                        .get_current_frame()
+                        .expect("Failed to acquire next swap chain texture")
+                        .output;
+                    let mut encoder = device
+                        .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+                    {
+                        let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                            color_attachments: &[wgpu::RenderPassColorAttachmentDescriptor {
+                                attachment: &frame.view,
+                                resolve_target: None,
+                                ops: wgpu::Operations {
+                                    load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
+                                    store: true,
+                                },
+                            }],
+                            depth_stencil_attachment: None,
+                        });
+                        rpass.set_pipeline(&render_pipeline);
+                        rpass.draw(0..3, 0..1);
+                    }
+
+                    queue.submit(Some(encoder.finish()));
+                }
+            }
+            Event::WindowEvent {
+                event: WindowEvent::CloseRequested,
+                ..
+            } => *control_flow = ControlFlow::Exit,
+            Event::WindowEvent {
+                event:
+                    WindowEvent::KeyboardInput {
+                        input:
+                            KeyboardInput {
+                                virtual_keycode: Some(VirtualKeyCode::Escape),
+                                ..
+                            },
+                        ..
+                    },
+                ..
+            } => *control_flow = ControlFlow::Exit,
+            _ => {}
+        }
+    });
+}
+
+#[cfg_attr(target_os = "android", ndk_glue::main(backtrace = "full"))]
+pub fn main() {
+    let event_loop = EventLoop::new();
+    let window = winit::window::WindowBuilder::new()
+        .with_title("Rust GPU - wgpu")
+        .with_inner_size(winit::dpi::LogicalSize::new(1280.0, 720.0))
+        .build(&event_loop)
+        .unwrap();
+
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        #[cfg(not(target_os = "android"))]
+        wgpu_subscriber::initialize_default_subscriber(None);
```

Comment:
looks like `wgpu_subscriber` is not used on android, could we cfg it out for android in Cargo.toml? (it's already cfg'd out for wasm32)

Reply:
This caused a crash on-device but I haven't looked at the problem yet. Would be nice to have some form of tracing when necessary I guess?

Reply:
The problem turns out to be wgpu attaching a log tracer which conflicts with the Android logger _if_ set up. That Android logger has intentionally been left out as we do not print any logs anywhere in this example, and we'd need to wait for https://github.com/rust-windowing/android-ndk-rs/pull/88 to land in order to not clutter `Cargo.toml` with some "unused" crate.